File: peernexus-frontend/src/services/authService.js
import apiClient from "./apiClient.js";

export const authService = {
  register: (payload) =>
    apiClient.post("/api/auth/register", payload).then((r) => r.data.data),

  login: (payload) =>
    apiClient.post("/api/auth/login", payload).then((r) => r.data.data),

  refreshToken: (refreshToken) =>
    apiClient.post("/api/auth/refresh", { refreshToken }).then((r) => r.data.data),
};

File: peernexus-backend/src/main/java/com/peernexus/peernexus/auth/service/AuthService.java
package com.peernexus.peernexus.auth.service;

import com.peernexus.peernexus.auth.dto.AuthResponse;
import com.peernexus.peernexus.auth.dto.LoginRequest;
import com.peernexus.peernexus.auth.dto.RefreshTokenRequest;
import com.peernexus.peernexus.auth.dto.RegisterRequest;
import com.peernexus.peernexus.auth.dto.ForgotPasswordRequest;
import com.peernexus.peernexus.auth.dto.ResetPasswordRequest;
import com.peernexus.peernexus.auth.dto.ResendVerificationRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refreshToken(RefreshTokenRequest request);

    void verifyEmail(String token);

    void resendVerificationEmail(ResendVerificationRequest request);

    void forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);

    void logout(RefreshTokenRequest request);
}
